Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -5113,29 +5113,71 @@ Vector<JS::Handle<DOM::Range>> Document::find_matching_text(String const& query,
     if (!document_element() || !document_element()->layout_node())
         return {};
 
-    Vector<JS::Handle<DOM::Range>> matches;
-    document_element()->layout_node()->for_each_in_inclusive_subtree_of_type<Layout::TextNode>([&](auto const& text_node) {
-        auto const& text = text_node.text_for_rendering();
-        size_t offset = 0;
-        while (true) {
-            auto match_index = case_sensitivity == CaseSensitivity::CaseInsensitive
-                ? text.find_byte_offset_ignoring_case(query, offset)
-                : text.find_byte_offset(query, offset);
-            if (!match_index.has_value())
-                break;
+    struct TextPositionNode : RefCounted<TextPositionNode> {
```

Comment:
Hmm, does this type really need reference counting?

Reply:
The reference counting was indeed unnecessary here. Removed.